cluster-operator/src/main/java/io/strimzi/operator/cluster/model/KafkaCluster.java

```diff
@@ -770,16 +770,19 @@ protected static void validateIntConfigProperty(String propertyName, KafkaCluste
 
     @SuppressWarnings("deprecation")
     static List<Toleration> tolerations(KafkaClusterSpec kafkaClusterSpec) {
+        List<Toleration> tolerations;
         if (kafkaClusterSpec.getTemplate() != null
                 && kafkaClusterSpec.getTemplate().getPod() != null
                 && kafkaClusterSpec.getTemplate().getPod().getTolerations() != null) {
             if (kafkaClusterSpec.getTolerations() != null) {
                 log.warn("Tolerations given on both spec.kafka.tolerations and spec.kafka.template.pod.tolerations; latter takes precedence");
             }
-            return kafkaClusterSpec.getTemplate().getPod().getTolerations();
+            tolerations = kafkaClusterSpec.getTemplate().getPod().getTolerations();
         } else {
-            return kafkaClusterSpec.getTolerations();
+            tolerations = kafkaClusterSpec.getTolerations();
         }
+        ModelUtils.removeEmptyValuesFromTolerations(tolerations);
+        return tolerations;
     }
```

Comment:
We have this method 4 times... I wonder if we can factor out a method in ModelUtils which takes a `PodTemplate` and a `List<Toleration>` in order to simplify the call sites to something like:

```java
return ModelUtils.tolerations("spec.kafka.template.pod.tolerations",
                              kafkaClusterSpec.getTemplate() != null ? kafkaClusterSpec.getTemplate().getPod() : null,
                              "spec.kafka.tolerations",
                              kafkaClusterSpec.getTolerations());
```

Reply:
I was thinking the same. Thanks for the confirmation.